### ISSUE TITLE
Environment variables not propagating as expected via `BB_ENV_PASSTHROUGH`

### ISSUE THREAD

**Issue by mjmartineau** [darknighte/bbsh#2]
I noticed that commands running via `bbsh` did not have access to my SSH agent, so git clone/fetch operations were failing due to lack of authentication.

I tried adding the relevant environment vars in a couple of places:

- `export BB_ENV_PASSTHROUGH='SSH_AUTH_SOCK SSH_AGENT_PID'`
- `export BB_ENV_PASSTHROUGH_ADDITIONS='SSH_AUTH_SOCK SSH_AGENT_PID'`

What worked was:

- `export BB_ENV_EXTRAWHITE='SSH_AUTH_SOCK SSH_AGENT_PID'`

**Comment by darknighte** [comment]
Good catch on these.  It is a sign of a bit of bit rot.  These variables changed at some point in the past in upstream Yocto Project, with the following mapping:
#   BB_ENV_WHITELIST  --> BB_ENV_PASSTHROUGH
#   BB_ENV_EXTRAWHITE --> BB_ENV_PASSTHROUGH_ADDITIONS

It may be a bit tricky to handle this for all cases since there is a bitbake check that will fail if the old value, BB_ENV_WHITELIST, propagates through.
